lib/withOnyx.js

```diff
@@ -88,7 +103,7 @@ export default function (mapOnyxToState) {
              * @param {*} val
              */
             setWithOnyxState(statePropertyName, val) {
-                if (!this.state.loading) {
+                if (!this.state.loading && this.state[statePropertyName] !== val) {
```

Comment:
Why this part?

```js
this.state[statePropertyName] !== val
```

Like, say we were able to set up the component in the constructor with all cache values. Loading is `false`. But if the value has changed we will start messing with `tempState` again? My expectation is that once we have all the values then we should be done with the other part of this method.

Reply:
you are right, i double checked it, and this part isn't necessary anymore!

Reply:
Great, let's remove it 😄

Reply:
wow it is gone GitHub trolled me 😂